cli/cli_state.ml

```diff
@@ -130,8 +132,8 @@ module Notify = struct
     Lwt_unix.openfile fd [flag_of_mode mode] 0 >|=
     Lwt_io.of_fd ~mode:mode
 
-  (* TODO(infinity0): perhaps log the exception somewhere *)
-  let catch_exn f = Lwt.catch f (fun _ -> Lwt.return_unit)
+  let catch_exn ui_mvar f = Lwt.catch f (fun e ->
+      selflog ui_mvar "async error in notify r/w" (Printexc.to_string e))
```

Comment:
does at no point the `state_mvar` get pushed by the loop reading `ui_mvar` (if this is the case, there is a potential deadlock (see https://github.com/hannesm/jackline/commit/90b8068adf218fd73e87f33c3cf57417bacdf0ec for a solution of a similar issue))